### PR TITLE
Greta/api routes and constants

### PR DESCRIPTION
**Pull request by gretahuang**
Updated `api routes`, added `api constants` file, added `request` and `recurrence` controllers (although `recurrence` controller is not implemented yet). @eric-ls

### CODE REVIEW

File: app/assets/javascripts/helpers/api_constants.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class ApiConstants {
+
+    get locations() {
+        return {
+            update     : (id) => `/api/locations/${id}`,
+            create : (id) => `/api/locations`
```
Comment (eric-ls) [inline app/assets/javascripts/helpers/api_constants.jsx]: nit: make sure these are aligned

File: app/assets/javascripts/recurrences.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
```
Comment (eric-ls) [inline app/assets/javascripts/recurrences.js]: can you remove this file

File: app/assets/javascripts/requests.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
```
Comment (eric-ls) [inline app/assets/javascripts/requests.js]: remove

File: app/assets/stylesheets/recurrences.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Recurrences controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (eric-ls) [inline app/assets/stylesheets/recurrences.scss]: same

File: app/assets/stylesheets/requests.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Requests controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (eric-ls) [inline app/assets/stylesheets/requests.scss]: same

File: app/assets/javascripts/helpers/api_constants.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+class ApiConstants {
+
+    get locations() {
```
Comment (warrenshen) [inline app/assets/javascripts/helpers/api_constants.jsx]: Two spaces would be nice :)

File: app/controllers/api/requests_controller.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+			render_json_message(:ok, message: 'Request successfully updated!')
+		else
+			render_json_message(:forbidden, errors: request.errors.full_messages)
+		end
+	end
+
+	private
+
+	def request_params
+	  params.require(:request).permit(:day, :start_time, :end_time, :frequency, :has_sent, :start_date,
```
Comment (warrenshen) [inline app/controllers/api/requests_controller.rb]: I'd format as follows:
```
params.require(:request).permit(
  :day,
  :start_time,
  ...
  :start_date,
)
```

Comment (warrenshen) [inline app/controllers/api/requests_controller.rb]: What does `exception` do?

File: config/routes.rb [hunk 8]
```diff
@@ -6,4 +6,17 @@
   root 'pages#home'
   get 'business' => 'pages#business'
   get 'styles' => 'pages#style'
+
+  resources :admins
+	resources :businesses
```
Comment (warrenshen) [inline config/routes.rb]: Spacing issues.

File: app/controllers/api/requests_controller.rb (hunk 7 above)
Comment (gretahuang) [inline app/controllers/api/requests_controller.rb]: Exception is for when people decide to cancel a food pickup request for `exception` # of pickups -- if `exception > 0`, don't add it to daily pickups and decrement exception.

**Comment by eric-ls** [comment]
LGTM! 👍 you did a greta job

**Pull request merged**